### PR TITLE
docs: update branding

### PR DESCRIPTION
**Pull request by desimone**
## Summary

Update docs to use the new logo, favicon, and remove some outdate sections.

## Related issues

https://github.com/pomerium/internal/issues/483

## Checklist

- [x] reference any related issues
- [x] updated docs
- [x] add appropriate tag (`improvement` / `bug` / etc)
- [x] ready for review

### CODE REVIEW

File: docs/docs/releases.md [hunk 1]
```diff
@@ -101,7 +101,7 @@ See the [README](https://github.com/pomerium/pomerium-helm/blob/master/charts/po
 
 ::: tip
 
-Officially supported build platforms are limited by [envoy proxy](https://www.envoyproxy.io/). If you have an enoy binary for your platform in your path at start time, `pomerium` should function correctly.
+Officially supported build platforms are limited by [envoy proxy](https://www.envoyproxy.io/). If you have an envoy binary for your platform in your path at start time, `pomerium` should function correctly.
```
Comment (travisgroth) [inline docs/docs/releases.md:104]: We actually don't support this presently.

**Review by travisgroth — CHANGES_REQUESTED** [review]
LGTM but one I think the tip for using your own envoy binary should be removed

File: docs/docs/releases.md (hunk 1 above)
Comment (desimone) [inline docs/docs/releases.md:104]:
```suggestion
Officially supported build platforms are limited by [envoy proxy](https://www.envoyproxy.io/). 
```

Comment (desimone) [reply docs/docs/releases.md]: Removed, LMK.

**Review by travisgroth — APPROVED** [review]

**Pull request merged**